src/main/kotlin/nmcp/NmcpAggregationExtension.kt

```diff
@@ -0,0 +1,71 @@
+package nmcp
+
+import gratatouille.GExtension
+import nmcp.internal.configureAttributes
+import nmcp.internal.nmcpConsumerConfigurationName
+import nmcp.internal.task.registerPublishTask
+import org.gradle.api.Action
+import org.gradle.api.Project
+import org.gradle.api.tasks.bundling.Zip
+
+@GExtension(pluginId = "com.gradleup.nmcp.aggregation")
+open class NmcpAggregationExtension(private val project: Project) {
+    internal val spec = project.objects.newInstance(NmcpSpec::class.java)
+
+    internal val consumerConfiguration = project.configurations.create(nmcpConsumerConfigurationName) {
+        it.isCanBeResolved = true
+        it.isCanBeConsumed = false
+
+        it.configureAttributes(project)
+    }
```

Comment:
Technically speaking, [Gradle does not recommend](https://docs.gradle.org/current/userguide/declaring_configurations.html#sec:configuration-flags-roles) using the same configuration for both dependency declaration and dependency consumption (see `canBeDeclared` flag)

Reply:
That means we need 2 configurations now, right? With the "resolvable" extending from the "declarable". I'm okay with that I guess, just feels less ergonomic so I won't rush into it.